cluster-autoscaler/cloudprovider/aws/auto_scaling_groups.go

```diff
@@ -308,13 +308,36 @@ func (m *asgCache) DeleteInstances(instances []*AwsInstanceRef) error {
 		}
 	}
 
+	// Initialize the success flag for recent scaling activity.
+	var recentScalingActivitySuccess = false
+	var err error
+
+	// Check if there are any placeholder instances in the list.
+	if m.HasPlaceholder(instances) {
+		// Log the check for placeholders in the ASG.
+		klog.V(4).Infof("Detected a placeholder instance, checking recent scaling activity for ASG %s", commonAsg.Name)
+
+		// Retrieve the most recent scaling activity to determine its success state.
+		recentScalingActivitySuccess, err = m.getMostRecentScalingActivity(commonAsg)
+
+		// Handle errors from retrieving scaling activity.
+		if err != nil {
+			// Log the error if the scaling activity check fails and return the error.
+			klog.Errorf("Error retrieving scaling activity for ASG %s: %v", commonAsg.Name, err)
+			return err // Return error to prevent further processing with uncertain state information.
+		}
+	}
+
 	for _, instance := range instances {
-		// check if the instance is a placeholder - a requested instance that was never created by the node group
-		// if it is, just decrease the size of the node group, as there's no specific instance we can remove
 		if m.isPlaceholderInstance(instance) {
-			klog.V(4).Infof("instance %s is detected as a placeholder, decreasing ASG requested size instead "+
-				"of deleting instance", instance.Name)
-			m.decreaseAsgSizeByOneNoLock(commonAsg)
+			if !recentScalingActivitySuccess {
```

Comment:
Hi @ruiscosta I don't believe this actually solves the problem.  Consider the following scenario:

1. We scale up the ASG by 10 instances.  AWS creates 3 of them, and then fails on the remaining 7.
2. `DeleteInstances` is called with those 10 instances; 7 of them are placeholders
3. In lines 321 we check to see if the most recent scaling activity was successful or not, which returns false since 7 instances could not be created.
4. Now for each instance in the loop, we decrease the ASG size by one, which reduces the ASG size by 7.
5. In between the check in line 321 and (say) the 5th iteration of the loop, AWS launches a new instance, which joins the cluster.  Our information about what instances are actually placeholders is now out of date, and we get the same problem that we had before.

We "could" check the recent scaling activity in every iteration of the loop, at the expense of making a _lot_ more API calls, which I think is undesirable, and is still subject to a race between when you make the check and when you change the ASG size.